plugins/module_utils/network/sonic/config/l2_acls/l2_acls.py

```diff
@@ -0,0 +1,566 @@
+        commands, requests = self.set_config(existing_l2_acls_facts)
+        if commands:
+            if not self._module.check_mode:
+                try:
+                    edit_config(self._module, to_request(self._module, requests))
+                except ConnectionError as exc:
+                    self._handle_failure_response(exc)
+
+            result['changed'] = True
+
+        changed_l2_acls_facts = self.get_l2_acls_facts()
+
+        result['before'] = existing_l2_acls_facts
+        if result['changed']:
+            result['after'] = changed_l2_acls_facts
+
+        result['commands'] = commands
+        result['warnings'] = warnings
+        return result
+
+    def set_config(self, existing_l2_acls_facts):
+        """ Collect the configuration from the args passed to the module,
+            collect the current configuration (as a dict from facts)
+
+        :rtype: A list
+        :returns: the commands necessary to migrate the current configuration
+                  to the desired configuration
+        """
+        want = self._module.params['config']
+        if want:
+            want = self.validate_and_normalize_config(want)
+        else:
+            want = []
+
+        have = existing_l2_acls_facts
+        resp = self.set_state(want, have)
+        return to_list(resp)
+
+    def set_state(self, want, have):
+        """ Select the appropriate function based on the state provided
+
+        :param want: the desired configuration as a dictionary
+        :param have: the current configuration as a dictionary
+        :rtype: A list
+        :returns: the commands necessary to migrate the current configuration
+                  to the desired configuration
+        """
+        state = self._module.params['state']
+        if state in ('merged', 'overridden', 'replaced'):
+            commands, requests = self._state_merged_overridden_replaced(want, have, state)
+        elif state == 'deleted':
+            commands, requests = self._state_deleted(want, have)
+
+        return commands, requests
+
+    def _handle_failure_response(self, connection_error):
+        log = None
+        try:
+            response = literal_eval(connection_error.args[0])
+            error_app_tag = response['ietf-restconf:errors']['error'][0].get('error-app-tag')
+        except Exception:
+            pass
+        else:
+            if error_app_tag == 'too-many-elements':
+                log = 'Exceeds maximum number of ACL / ACL Rules'
+            elif error_app_tag == 'update-not-allowed':
+                log = 'Creating ACLs with same name and different type not allowed'
+
+        if log:
+            response.update({u'log': log})
+            self._module.fail_json(msg=to_text(response), code=connection_error.code)
+        else:
+            self._module.fail_json(msg=str(connection_error), code=connection_error.code)
+
+    def _state_merged_overridden_replaced(self, want, have, state):
```

Comment:
For merged, for same acl, if "have" has more attributes or sub-container configuration than "want",  are they deleted?
If possible, can some regression test cases be added!

Reply:
Each L2 ACL rule is either created or deleted at once and is not modified.
In state merged, for an existing sequence number in a L2 ACL, if have and want attributes are not equal (different attributes, different values for same attributes) then the module execution will be aborted with the error message "Cannot update existing sequence `<seq num>` of L2 ACL `<acl name>` with state merged. Please use state replaced or overridden."

Reply:
Thanks for explaination!